include/range/v3/view/any_view.hpp

```diff
@@ -31,14 +31,80 @@ namespace ranges
 {
     inline namespace v3
     {
+        /// \brief An enum that denotes the supported subset of range concepts supported by a range.
         enum class category
         {
-            input,
-            forward,
-            bidirectional,
-            random_access
+            none            = 0b00000000,  ///<\brief No concepts met.
+            input           = 0b00000001,  ///<\brief Satisfies ranges::v3::concepts::InputRange
+            forward         = 0b00000011,  ///<\brief Satisfies ranges::v3::concepts::ForwardRange
+            bidirectional   = 0b00000111,  ///<\brief Satisfies ranges::v3::concepts::BidirectionalRange
+            random_access   = 0b00001111,  ///<\brief Satisfies ranges::v3::concepts::RandomAccessRange
+            sized           = 0b00010000   ///<\brief Satisfies ranges::v3::concepts::SizedRange
         };
 
+        /** \name Binary operators for ranges::v3::category
+         *  \relates ranges::v3::category
+         *  \{
+         */
+        constexpr category operator& (category lhs, category rhs)
+        {
+            return static_cast<category>(static_cast<std::underlying_type_t<category>>(lhs) & static_cast<std::underlying_type_t<category>>(rhs));
+        }
+
+        constexpr category operator| (category lhs, category rhs)
+        {
+            return static_cast<category>(static_cast<std::underlying_type_t<category>>(lhs) | static_cast<std::underlying_type_t<category>>(rhs));
+        }
+
+        constexpr category operator^ (category lhs, category rhs)
+        {
+            return static_cast<category>(static_cast<std::underlying_type_t<category>>(lhs) ^ static_cast<std::underlying_type_t<category>>(rhs));
+        }
+
+        constexpr category operator~ (category lhs)
+        {
+            return static_cast<category>(~static_cast<std::underlying_type_t<category>>(lhs));
+        }
+
+        constexpr category & operator&= (category & lhs, category rhs)
+        {
+            lhs = lhs & rhs;
+            return lhs;
+        }
+
+        constexpr category & operator|= (category & lhs, category rhs)
+        {
+            lhs = lhs | rhs;
+            return lhs;
+        }
+
+        constexpr category & operator^= (category & lhs, category rhs)
+        {
+            lhs = lhs ^ rhs;
+            return lhs;
+        }
+        //!\}
+
+        /// \brief For a given range, return a ranges::v3::category enum with the satisfied concepts.
+        template<typename Rng>
+        constexpr category get_categories(Rng &&)
```

Comment:
> This function also is not C++11 constexpr, and it needs to be so. I suggest bringing back the old to_cat_ overloads,

I have just made it c++11 compatible.

> It also cannot take the range as a function parameter: parameters of constexpr functions must have literal type, so this function is only constexpr when Rng is literal. Callers will have to explicitly specify the template parameter.

Hm, not sure I get this. Reference types are always literal or not? And Rng is always either & or &&. The tests also seem to work without explicitly specifying the type...

Reply:
> Hm, not sure I get this.

I'm not surprised, I've garbled the comment completely. The real issue is that `get_categories(Rng{})` is only a constant expression when `Rng` is default constructible *literal* type, whereas `get_categories<Rng>()` could be formulated to have no such problem.

Reply:
> The real issue is that get_categories(Rng{}) is only a constant expression when Rng is a default constructible literal type, 

Yep, that makes sense. Rng types might not be constexpr default constructible.

Reply:
It does however reduce usability in the cases where people want
```cpp
any_view<int, get_categories(some_range)> foo;
```
Because they then need to add decltype... but then again this will become better with c++17 and type deduction.

Reply:
Having to write foo<bar>() really hurts the eye and screams for variable templates 😢